### PR TITLE
VVVVVV: Make unnecessary Trinkets filler

### PR DESCRIPTION
**Pull request by Scrungip**
## What is this fixing or adding?
Should clear up confusion with VVVVVV's trinkets by making the non-required ones filler.

## How was this tested?
Ran some test generations, tested with all possible values for Door Cost.

### CODE REVIEW

**Review by N00byKing — APPROVED** [review]
This used to cause generation issues. If that is no longer the case, lgtm

File: worlds/v6/__init__.py [hunk 1]
```diff
@@ -49,12 +49,14 @@ def set_rules(self):
         self.area_cost_map = {}
         set_rules(self.multiworld, self.options, self.player, self.area_connections, self.area_cost_map)
 
-    def create_item(self, name: str) -> Item:
-        return V6Item(name, ItemClassification.progression, item_table[name], self.player)
+    def create_item(self, name: str, classification: ItemClassification) -> Item:
```
Comment (NewSoupVi) [inline worlds/v6/__init__.py:52]: This is a function that's overridden from the superclass (World), and your new function signature is incompatible with the one from the superclass.

You can fix this by supplying a default value for `classification`, which should ideally be "filler":

`def create_item(self, name: str, classification: ItemClassification = ItemClassification.filler) -> Item:`

**Review by NewSoupVi — CHANGES_REQUESTED** [review]

**Comment by NewSoupVi** [comment]
Also, I don't think anything has significantly changed in core to fix the generation errors this would cause, but we can see.

File: worlds/v6/__init__.py (hunk 1 above)
Comment (Scrungip) [reply worlds/v6/__init__.py:52]: Got it, I'll add that in tonight.

**Comment by Scrungip** [comment]
I didn't see any generation errors when I was testing, but I might have just been lucky.

File: worlds/v6/__init__.py (hunk 1 above)
Comment (Exempt-Medic) [inline worlds/v6/__init__.py:52]:
```suggestion
    def create_item(self, name: str, classification: ItemClassification = ItemClassification.filler) -> Item:
```

File: worlds/v6/__init__.py [hunk 2]
```diff
@@ -49,12 +49,14 @@ def set_rules(self):
...
+        return V6Item(name, classification, item_table[name], self.player)
 
     def create_items(self):
-        trinkets = [self.create_item("Trinket " + str(i+1).zfill(2)) for i in range(0,20)]
-        self.multiworld.itempool += trinkets
+        progtrinkets = [self.create_item("Trinket " + str(i+1).zfill(2), ItemClassification.progression) for i in range(0, (4 * self.options.door_cost.value))]
+        filltrinkets = [self.create_item("Trinket " + str(i+1).zfill(2), ItemClassification.filler) for i in range((4 * self.options.door_cost.value), 20)]
```
Comment (Exempt-Medic) [inline worlds/v6/__init__.py:57]:
```suggestion
        filltrinkets = [self.create_item("Trinket " + str(i+1).zfill(2)) for i in range((4 * self.options.door_cost.value), 20)]
```

**Review by Exempt-Medic — CHANGES_REQUESTED** [review]
With these changes, I did five thousand random solo generations and did not run into any errors. I also checked that the correct trinkets were being made progression and still being required properly by logic (ya know, just in case)

**Review by NewSoupVi — APPROVED** [review]

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**